### PR TITLE
Introduce SpeziPersonalInfo target

### PR DESCRIPTION
**Pull request by Supereg**
# Introduce SpeziPersonalInfo target

## :recycle: Current situation & Problem
Previously, `SpeziViews` was treated like a catch all for any UI-component-based API interfaces. This PR introduces `SpeziPersonInfo` that paves the way for a more granular grouping of UI components. In this PR we move out any UI components that deal with personal information. There might be future additions coming from a SpeziAccount refactoring.

This brings some breaking changes, where we removed the `NameFields` view and instead replaced it with new optimized views `NameTextField` and `NameFieldRow`. These address specific parts of a `PersonNameComponents` and remove a lot of the complexity like focus state handling. As you now can directly access the individual TextFields, this can be more cleanly handled by the users themselves.

Further this PR removes the unused `HTMLView` (and `DocumentView`). `MarkdownView` remains.

This PR updates the project to target a deployment target of iOS 17 and migrates to use the new String Catalogs. We made sure to use the explicit `init(verbatim:)` initializer for `Text` views where necessary. We exposed similar functionality with our `LazyText` und `Label` views. External parameters were changed and induce a breaking change.

## :gear: Release Notes 
* The new `SpeziPersonalInfo` target is the entry point for all View components that deal with personal information.
* The `NameFields` view was removed and replaced with row-based views `NameTextField` and `NameFieldRow`.
* The `HTMLView` and `DocumentView` were removed.
* New initializer argument labels for `LazyText` and `Label` views.
* Added Documentation Catalogs to structure documentation more clearly.
* Upgraded to use String Catalogs.
* Updated target platform to iOS 17.

## :books: Documentation
The PR adds documentation catalogs to both targets which were previously missing. Through these documentation catalogs we create a structure that allows to more easily explore SpeziViews and SpeziPersonalInfo packages. Certain documentation was optimized to more clearly communicate the technical details.

## :white_check_mark: Testing
Testing was update to separate functionality of the two new targets to provide a clear separation between both.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Supereg** [comment]
@PSchmiedmayer are there any other changes we want to bring to SpeziViews?

File: Sources/SpeziViews/Views/Text/LazyText.swift [hunk 1]
```diff
@@ -9,27 +9,33 @@
 import SwiftUI
 
 
+private struct TextLine: Identifiable {
+    var id: UUID
+    var line: String
+}
```
Comment (PSchmiedmayer) [inline Sources/SpeziViews/Views/Text/LazyText.swift:15]: Nice, thank you!

File: Sources/SpeziViews/Views/Text/TextContent.swift
Comment (PSchmiedmayer) [inline Sources/SpeziViews/Views/Text/TextContent.swift:1]: 👍

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements, looks great! 👍

**Comment by vishnuravi** [comment]
Nice work @Supereg!

**Comment by PSchmiedmayer** [comment]
> @PSchmiedmayer are there any other changes we want to bring to SpeziViews?

Oh, sorry, missed the message here. The main thing would be the re-organization that we alluded to in our meeting yesterday but that might be a separate PR but maybe something we address before we tag a new breaking release?

**Comment by Supereg** [comment]
@PSchmiedmayer as discussed I extended the scope of the PR to start with splitting SpeziViews into multiple targets. As this substantially changed the scope of the PR, I just re-requested a review from you. Looking through the updated PR description should give you a good overview of the updated scope.

File: Sources/SpeziPersonalInfo/Fields/NameTextField.swift [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+///
+/// - Note: A empty string will be automatically mapped to a `nil` value for the respective property of `PersonNameComponents`.
+public struct NameTextField<Label: View>: View {
+    private let prompt: Text?
+    private let label: Label
+    private let nameComponent: WritableKeyPath<PersonNameComponents, String?>
+
+    @Binding private var name: PersonNameComponents
+
+    private var componentBinding: Binding<String> {
```
Comment (PSchmiedmayer) [inline Sources/SpeziPersonalInfo/Fields/NameTextField.swift:38]: Nice way to do this!

File: Sources/SpeziViews/SpeziViews.docc/SpeziViews.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# ``SpeziViews``
+
+A Spezi framework that provides a common set of SwiftUI views and related functionality used across the Spezi ecosystem.
```
Comment (PSchmiedmayer) [inline Sources/SpeziViews/SpeziViews.docc/SpeziViews.md:3]: Nice summary of the different functions!

File: Tests/UITests/UITests.xcodeproj/xcshareddata/xcschemes/TestApp.xcscheme [hunk 4]
```diff
@@ -37,7 +37,7 @@
       </BuildActionEntries>
    </BuildAction>
    <TestAction
-      buildConfiguration = "Debug"
+      buildConfiguration = "Release"
```
Comment (PSchmiedmayer) [inline Tests/UITests/UITests.xcodeproj/xcshareddata/xcschemes/TestApp.xcscheme:40]: Not sure if this is intentional here to use release builds for testing.

**Review by PSchmiedmayer — APPROVED** [review]
Looks great, thank you for the work here!

File: Tests/UITests/UITests.xcodeproj/xcshareddata/xcschemes/TestApp.xcscheme (hunk 4 above)
Comment (Supereg) [reply Tests/UITests/UITests.xcodeproj/xcshareddata/xcschemes/TestApp.xcscheme]: Ah, what a misclick. Intention was to change it to "Testing" 😅

**Pull request merged**